File: node-ts/src/estimate.ts
export type EstimateParams = {
  area: number;
  unitPrice: number;
};

export function calculateEstimate(params: EstimateParams): number {
  const { area, unitPrice } = params;

  if (!Number.isFinite(area) || !Number.isFinite(unitPrice)) {
    throw new Error("Invalid input type");
  }
  if (area <= 0 || unitPrice <= 0) {
    throw new Error("Values must be greater than zero");
  }

  return area * unitPrice;
}

File: go/estimate/estimate.go
package estimate

import "fmt"

type Params struct {
	Area      float64
	UnitPrice float64
}

func Calculate(p Params) (float64, error) {
	if p.Area <= 0 || p.UnitPrice <= 0 {
		return 0, fmt.Errorf("values must be greater than zero")
	}
	return p.Area * p.UnitPrice, nil
}
